src/components/policy/policy_external/src/sql_pt_representation.cc

```diff
@@ -616,19 +616,32 @@ bool SQLPTRepresentation::GatherFunctionalGroupings(
     LOG4CXX_WARN(logger_, "Incorrect select from functional_groupings");
     return false;
   }
+
   utils::dbms::SQLQuery rpcs(db());
   if (!rpcs.Prepare(sql_pt::kSelectAllRpcs)) {
     LOG4CXX_WARN(logger_, "Incorrect select all from rpc");
     return false;
   }
 
+  utils::dbms::SQLQuery external_consent_entities(db());
+  if (!external_consent_entities.Prepare(
+          sql_pt::kSelectExternalConsentEntities)) {
+    LOG4CXX_WARN(logger_,
+                 "Incorrect select statement for 'external_consent_entities'.");
+    return false;
+  }
+
   while (func_group.Next()) {
     policy_table::Rpcs rpcs_tbl;
+
     if (!func_group.IsNull(2)) {
       *rpcs_tbl.user_consent_prompt = func_group.GetString(2);
     }
-    int func_id = func_group.GetInteger(0);
-    rpcs.Bind(0, func_id);
+
+    const int group_id = func_group.GetInteger(0);
+
+    rpcs.Bind(0, group_id);
```

Comment:
@LuxoftAKutsan More or less it is a subjective matter and it is hard to be given the right answer, right?